extensions/mssql/src/objectManagement/objectManagementService.ts

```diff
@@ -83,11 +83,22 @@ export class ObjectManagementService extends BaseService implements IObjectManag
 		return this.runWithErrorHandling(contracts.AttachDatabaseRequest.type, params);
 	}
 
+	async getRestorePlan(restoreParams: RestoreParams): Promise<RestorePlanResponse> {
+		const params: contracts.RestoreParams = restoreParams;
+		return this.runWithErrorHandling(contracts.RestorePlanRequest.type, params);
```

Comment:
You shouldn't need to be redefining the message contracts here - we already have that done in sqlops-dataprotocolclient. So you can instead just get the provider using `azdata.dataprotocol.getProvider<azdata.RestoreProvider>('MSSQL', azdata.DataProviderType.RestoreProvider)` and then call the functions on it and pass the params to that instead.

@corivera FYI, since it looks like you did the same for backup and I just missed that apparently. Did you have a reason for redefining the stuff here?

Reply:
Thank you, I have used the provider to make the required calls and removed the duplicated code.

Reply:
Tested everything and looks good!